### PR TITLE
Upgrade protobuf to 1.4.0

### PR DESCRIPTION
**Pull request by johanbrandhorst**

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Yeah so this might be a no-go until we can do a v2. Basically all the proto packages in the public API have incompatible changes, and some function signatures are now incompatible:
https://app.circleci.com/jobs/github/grpc-ecosystem/grpc-gateway/5804
```
github.com/grpc-ecosystem/grpc-gateway/runtime
----------------------------------------------
Incompatible changes:
- (*ServeMux).GetForwardResponseOptions: changed from func() []func(context.Context, net/http.ResponseWriter, github.com/golang/protobuf/proto.Message) error to func() []func(context.Context, net/http.ResponseWriter, google.golang.org/protobuf/runtime/protoiface.MessageV1) error
- ForwardResponseMessage: changed from func(context.Context, *ServeMux, Marshaler, net/http.ResponseWriter, *net/http.Request, github.com/golang/protobuf/proto.Message, ...func(context.Context, net/http.ResponseWriter, github.com/golang/protobuf/proto.Message) error) to func(context.Context, *ServeMux, Marshaler, net/http.ResponseWriter, *net/http.Request, google.golang.org/protobuf/runtime/protoiface.MessageV1, ...func(context.Context, net/http.ResponseWriter, google.golang.org/protobuf/runtime/protoiface.MessageV1) error)
- ForwardResponseStream: changed from func(context.Context, *ServeMux, Marshaler, net/http.ResponseWriter, *net/http.Request, func() (github.com/golang/protobuf/proto.Message, error), ...func(context.Context, net/http.ResponseWriter, github.com/golang/protobuf/proto.Message) error) to func(context.Context, *ServeMux, Marshaler, net/http.ResponseWriter, *net/http.Request, func() (google.golang.org/protobuf/runtime/protoiface.MessageV1, error), ...func(context.Context, net/http.ResponseWriter, google.golang.org/protobuf/runtime/protoiface.MessageV1) error)
- PopulateFieldFromPath: changed from func(github.com/golang/protobuf/proto.Message, string, string) error to func(google.golang.org/protobuf/runtime/protoiface.MessageV1, string, string) error
- PopulateQueryParameters: changed from func(github.com/golang/protobuf/proto.Message, net/url.Values, *github.com/grpc-ecosystem/grpc-gateway/utilities.DoubleArray) error to func(google.golang.org/protobuf/runtime/protoiface.MessageV1, net/url.Values, *github.com/grpc-ecosystem/grpc-gateway/utilities.DoubleArray) error
- QueryParameterParser.Parse: changed from func(github.com/golang/protobuf/proto.Message, net/url.Values, *github.com/grpc-ecosystem/grpc-gateway/utilities.DoubleArray) error to func(google.golang.org/protobuf/runtime/protoiface.MessageV1, net/url.Values, *github.com/grpc-ecosystem/grpc-gateway/utilities.DoubleArray) error
- StreamError.XXX_NoUnkeyedLiteral: removed
- StreamError.XXX_sizecache: removed
- StreamError.XXX_unrecognized: removed
- WithForwardResponseOption: changed from func(func(context.Context, net/http.ResponseWriter, github.com/golang/protobuf/proto.Message) error) ServeMuxOption to func(func(context.Context, net/http.ResponseWriter, google.golang.org/protobuf/runtime/protoiface.MessageV1) error) ServeMuxOption
```
etc.

**Comment by johanbrandhorst** [comment]
Rebased on v2

File: runtime/internal/examplepb/example.proto [hunk 1]
```diff
@@ -1,115 +1,104 @@
...
-	repeated string repeated_string_annotation = 34;
-
-	// repeated nested object comment (This comment is overridden by the field annotation)
-	repeated Nested repeated_nested_annotation = 35;
-
-	// nested object comments (This comment is overridden by the field annotation)
-	Nested nested_annotation = 36;
-
-	int64 int64_override_type = 37;
+  // Nested is nested type.
```
Comment (achew22) [inline runtime/internal/examplepb/example.proto:16]: Could we run `clang-format` on these? If this is the output from clang format then LGTM.

Comment (johanbrandhorst) [reply runtime/internal/examplepb/example.proto:14]: Formatted. I'll add a clang-format CI job for the protofiles in a follow up PR

Comment (johanbrandhorst) [reply runtime/internal/examplepb/example.proto:14]: Scratch that last part, clang-format ruins the beauty of `a_little_bit_of_everything.proto`.

**Comment by johanbrandhorst** [comment]
@achew22 @drigz can't understand why bazel is broken, the two recent errors seem in conflict:

```
INFO: Build completed successfully, 50 total actions
--- runtime/BUILD.bazel	1970-01-01 00:00:00.000000000 +0000
+++ runtime/BUILD.bazel	1970-01-01 00:00:00.000000000 +0000
@@ -66,11 +66,11 @@
     deps = [
         "//internal:go_default_library",
         "//utilities:go_default_library",
-        "//runtime/internal/examplepb:go_default_library",
         "@com_github_golang_protobuf//jsonpb:go_default_library_gen",
         "@com_github_golang_protobuf//proto:go_default_library",
         "@com_github_golang_protobuf//ptypes:go_default_library_gen",
         "@com_github_google_go_cmp//cmp:go_default_library",
+        "@com_github_grpc_ecosystem_grpc_gateway_v2//runtime/internal/examplepb:go_default_library",
         "@go_googleapis//google/api:httpbody_go_proto",
         "@go_googleapis//google/rpc:errdetails_go_proto",
         "@io_bazel_rules_go//proto/wkt:duration_go_proto",
ERROR: Bazel files out-of-date, please run `bazel run :gazelle
```
https://app.circleci.com/pipelines/github/grpc-ecosystem/grpc-gateway/168/workflows/f798c1c0-14e0-4053-b8cc-762cdeb315c7/jobs/6748

```
Analyzing: 61 targets (27 packages loaded, 0 targets configured)
ERROR: /src/grpc-gateway/runtime/BUILD.bazel:47:1: no such package '@com_github_grpc_ecosystem_grpc_gateway_v2//runtime/internal/examplepb': The repository '@com_github_grpc_ecosystem_grpc_gateway_v2' could not be resolved and referenced by '//runtime:go_default_test'
ERROR: Analysis of target '//runtime:go_default_test' failed; build aborted: no such package '@com_github_grpc_ecosystem_grpc_gateway_v2//runtime/internal/examplepb': The repository '@com_github_grpc_ecosystem_grpc_gateway_v2' could not be resolved
INFO: Elapsed time: 0.378s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (30 packages loaded, 10 targets configured)

```
https://app.circleci.com/pipelines/github/grpc-ecosystem/grpc-gateway/171/workflows/24f87f86-75cc-44bd-a0bb-58e4c7b28059/jobs/6770

Any ideas?

**Comment by achew22** [comment]
@johanbrandhorst I would bet you it is because Gazelle needs to be told the correct new prefix.

https://github.com/grpc-ecosystem/grpc-gateway/blob/8166034d5510ef987b9ae8c56099fa5cd235a0fb/BUILD#L16

**Comment by johanbrandhorst** [comment]
> @johanbrandhorst I would bet you it is because Gazelle needs to be told the correct new prefix.
> 
> https://github.com/grpc-ecosystem/grpc-gateway/blob/8166034d5510ef987b9ae8c56099fa5cd235a0fb/BUILD#L16

I thought I updated this on v2 already :thinking:

**Comment by johanbrandhorst** [comment]
Yeah that has already been fixed on v2: https://github.com/grpc-ecosystem/grpc-gateway/blob/v2/BUILD#L16

**Comment by johanbrandhorst** [comment]
@achew22 @drigz I'd still really appreciate help figuring out the bazel error. I don't understand what is wrong.

**Comment by drigz** [comment]
Looks like the Go importpath inside runtime/internal/examplepb is incorrect. https://github.com/drigz/grpc-gateway/commit/f15ae7fff203fad8ba37c1f09c3832be01725e66

**Comment by johanbrandhorst** [comment]
> Looks like the Go importpath inside runtime/internal/examplepb is incorrect. [drigz@f15ae7f](https://github.com/drigz/grpc-gateway/commit/f15ae7fff203fad8ba37c1f09c3832be01725e66)

Thank you so much Rodrigo! I'll fix it.

**Pull request merged**